setuptools_scm/utils.py

```diff
@@ -107,3 +107,53 @@ def has_command(name):
     if not res:
         warnings.warn("%r was not found" % name)
     return res
+
+
+if IS_WINDOWS:
+    from ctypes import create_unicode_buffer, windll, WinError
+    from ctypes.wintypes import MAX_PATH, LPCWSTR, LPWSTR, DWORD
+
+    GetLongPathNameW = windll.kernel32.GetLongPathNameW
+    GetLongPathNameW.argtypes = [LPCWSTR, LPWSTR, DWORD]
+    GetLongPathNameW.restype = DWORD
+
+    def get_windows_long_path_name(path):
+        """
+        Converts the specified path from short (MS-DOS style) to long form
+        using the 'GetLongPathNameW' function from Windows API.
+
+        https://msdn.microsoft.com/en-us/library/windows/desktop/aa364980(v=vs.85).aspx
+        """
+        if PY2:
+            # decode path using filesystem encoding on python2; on python3
+            # it is already a unicode string
+            path = unicode(path, sys.getfilesystemencoding())  # noqa
+
+        pathlen = MAX_PATH + 1
+        if DEBUG:
+            # test reallocation logic
+            pathlen = 1
+
+        for _ in range(2):
+            buf = create_unicode_buffer(pathlen)
+            retval = GetLongPathNameW(path, buf, pathlen)
+
+            if retval == 0:
+                # if the function fails for any reason (e.g. file does not
+                # exist), the return value is zero
+                raise WinError()
```

Comment:
Search for the word "WinError" in https://docs.python.org/3/library/ctypes.html and look at the example there.